File: public/cses/introductory-problems/repetitions/approach_2/repetitions.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // setup
    // setup-start
#ifndef ONLINE_JUDGE
    freopen("../input.txt", "r", stdin);
    freopen("../output.txt", "w", stdout);
#endif
    // setup-end

    string s;
    cin >> s;
    int n = s.size();
    int last_index = 0;
    int max_same = 0;
    for (int i = 1; i < n; i++) {
        if (s[i] != s[i - 1]) {
            max_same = max(max_same, i - last_index);
            last_index = i;
        }
    }
    max_same = max(max_same, n - last_index);
    cout << max_same << endl;
}

File: public/cses/introductory-problems/repetitions/approach_2/repetitions.py
# setup-start
import os
import sys
script_directory = os.path.dirname(os.path.realpath(__file__))
parent_directory = os.path.dirname(script_directory)
input_path = os.path.join(parent_directory, 'input.txt')
output_path = os.path.join(parent_directory, 'output.txt')
if os.path.exists(input_path):
    sys.stdin = open(input_path, 'r')
    try:
        sys.stdout = open(output_path, 'w')
    except Exception as e:
        pass
# setup-end


def main():
    s = input()
    n = len(s)
    last_index = 0
    max_same = 0

    for i in range(1, n):
        if s[i] != s[i - 1]:
            max_same = max(max_same, i - last_index)
            last_index = i

    max_same = max(max_same, n - last_index)
    print(max_same)


main()
